### PR TITLE
Rename to Draupnir in Appropriate Places

### PR DESCRIPTION
**Pull request by FSG-Cat**
This PR aims to make it so its harder to find places where the bot calls it self Mjolnir without it affecting code in ways that i cant begin to predict.

### CODE REVIEW

File: src/appservice/cli.ts
Comment (Gnuxie) [inline src/appservice/cli.ts:22]: This will surely break existing deployments?

File: Dockerfile
Comment (Gnuxie) [inline Dockerfile:24]: I'd prefer to be conservative about changing this, just in case someone else has built their own image based on our own.

File: src/appservice/cli.ts
Comment (FSG-Cat) [reply src/appservice/cli.ts:22]: it wont affect the playbook at all.

File: Dockerfile
Comment (FSG-Cat) [reply Dockerfile:24]: You do make a fair point about this. But i also do think that we should break this at some point as we should remove all references to mjolnir from the codebase that are not historical references.

As in if we refer to mjolnir it should be like for example refering to a Mjolnir issue or that this file integrates work from mjolnir.

Comment (Gnuxie) [reply Dockerfile:24]: > But i also do think that we should break this at some point as we should remove all references to mjolnir from the codebase that are not historical references.

Why?

File: src/appservice/cli.ts
Comment (Gnuxie) [reply src/appservice/cli.ts:22]: > Like im not sure who we would break by renaming the Reg file considering your explicitly passing the reg file location to the AS when running AS mode

Ok

File: Dockerfile
Comment (FSG-Cat) [reply Dockerfile:24]: it gives a certain vibe to some people that we refer to our selfs as mjolnir all over the place. This PR should hopefully reduce the chances of end users seeing us call our selfs mjolnir in logs or via command errors.

**Comment by Gnuxie** [comment]
Willing to give this the all clear, @MTRNord, do you have any thoughts before we merge this?

File: .gitignore
Comment (MTRNord) [inline .gitignore:10]: (just picking a random line of this change)

Make sure this is in the release notes please as it is potentially a breaking change and would break bots if not updated downstream

**Review by MTRNord — APPROVED** [review]
looks good to me

File: .gitignore
Comment (Gnuxie) [reply .gitignore:10]: Would it break bots? As far as we know, the path is never referenced directly or implicitly, you have to explicitly provide the path to the registration file to draupnir in the cli, and this is just to generate -- hmm.

@FSG-Cat does the documentation make reference to `mjolnir-registration.yaml` and ask you to move / copy it?

Comment (FSG-Cat) [reply .gitignore:10]: Docs never refers to any file called Draupnir or mjolnir that is pre existing. The file that we talk about in docs is created by Draupnir on demand with a file name of our choosing or the docs are wrong currently. (Remember D4A mdad and K8s both use synthetic Reg files that are completely made up by the maintainers of said deployment methods and not generated by Draupnir)

Comment (FSG-Cat) [reply .gitignore:10]: This file as far as i can tell is a file that mxtester needs but well i accounted for that.

The more likely to break file would be our entrypoint file if someone is refering to that for some reason. But thats already accounted for as a change that we need to tell people about.

Comment (Gnuxie) [reply .gitignore:10]: @FSG-Cat Could you add a `CHANGELOG.md` from the template https://github.com/Gnuxie/tsconfig/blob/main/templates/CHANGELOG.md

Add details for both the entrypoint change and the generated registration file change.

Then at the bottom:

```
## Versions v2.0.0-beta.7 and prior

Please see  https://github.com/the-draupnir-project/Draupnir/releases
```

I don't know if we will use this but it's going to at least help us keep unreleased changes in check.

Comment (FSG-Cat) [reply .gitignore:10]: I think its a great idea.

Comment (MTRNord) [reply .gitignore:10]: Turns out I _thought_ it was in https://github.com/the-draupnir-project/Draupnir/blob/MTRNord/draupnir4all/docs/appservice.md but a) it wasnt and b) I already used draupnir there...

File: CHANGELOG.md
Comment (Gnuxie) [inline CHANGELOG.md:14]: @FSG-Cat  you didn't write anything

File: CHANGELOG.md
Comment (Gnuxie) [inline CHANGELOG.md:20]: You should have wrote this in the unreleased part

File: CHANGELOG.md
Comment (Gnuxie) [inline CHANGELOG.md:22]:
```suggestion
## [Unreleased] - 2024-10-04

### Changed

* `Dockerfile`: entry-point was renamed from `mjolnir-entrypoint.sh` to `draupnir-entrypoint.sh`. If you have built a Dockerfile based on ours, you may need to make some changes.

* `Dockerfile`: source code was moved from `/mjolnir` to `/draupnir`. If you have built a custom docker image based on our Dockerfile based on ours, you may need to make some changes.

* The appservice registration file generator no longer emits `mjolnir-registration.yaml` as it has been renamed to `draupnir-registration.yaml`. This is only a concern if you have automated tooling that generates a registration file.

## Versions v2.0.0-beta.7 and prior

Please see https://github.com/the-draupnir-project/Draupnir/releases for more information.
```

**Review by Gnuxie — APPROVED** [review]

**Pull request merged**
